src/hotspot/share/gc/g1/g1CollectedHeap.cpp

```diff
@@ -511,172 +511,119 @@ HeapWord* G1CollectedHeap::attempt_allocation_slow(size_t word_size) {
   return NULL;
 }
 
-bool G1CollectedHeap::check_archive_addresses(MemRegion* ranges, size_t count) {
-  assert(ranges != NULL, "MemRegion array NULL");
-  assert(count != 0, "No MemRegions provided");
+bool G1CollectedHeap::check_archive_addresses(MemRegion range) {
   MemRegion reserved = _hrm.reserved();
-  for (size_t i = 0; i < count; i++) {
-    if (!reserved.contains(ranges[i].start()) || !reserved.contains(ranges[i].last())) {
-      return false;
-    }
+  if (!reserved.contains(range.start()) || !reserved.contains(range.last())) {
+    return false;
+  } else {
+    return true;
   }
```

Comment:
Maybe make this `return reserved.contains(range.start()) && reserved.contains(range.last())`

Reply:
Fixed.